File: java/src/main/java/ai/buble/sdk/Envelope.java
package ai.buble.sdk;

/**
 * Common response envelope used by Buble media, file, and app endpoints.
 *
 * @param <T> response data type
 */
public class Envelope<T> {
    private T data;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: dotnet/src/Buble.Sdk/Envelope.cs
using System.Text.Json.Serialization;

namespace Buble.Sdk;

/// <summary>
/// Standard envelope returned by Buble media, file, and app endpoints.
/// </summary>
public sealed class Envelope<T>
{
    [JsonPropertyName("data")]
    public T? Data { get; set; }
}
